doc/man7/openssl-threads.pod

```diff
@@ -0,0 +1,66 @@
+=pod
+
+=head1 NAME
+
+openssl-threads - Using multiple threads with OpenSSL
+
+=head1 DESCRIPTION
+
+OpenSSL can be built with or without threads support. The most important
+use of this support is so that OpenSSL itself can use a single consistent
+API, as shown in L<CRYPTO_THREAD_run_once(3)/EXAMPLES>.
+Multi-platform applications can also use this API.
+
+In particular, being configured for threads-support does not imply that
+all objects can be used simultaneously in more than one thread.
+To repeat: I<most objects are not safe for simultaneous use>.
+Exceptions to this should be documented on the specific manual pages, and
+some general high-level guidance is given here.
+
```

Comment:
I think there should be some discussion about calling functions from multiple threads, but using *different* objects, i.e. in most cases it is safe to call the same function in multiple threads as long as you use a different object.

There may be some exceptions to this rule, e.g. in some cases where global state is modified (OBJ_new_nid() and OBJ_create() spring to mind as instances which are not thread safe at all (this might actually be considered a bug for these functions))

Reply:
I added a couple of paragraphs in the new commit.